src/modules/commander/failure_detector/FailureDetector.cpp

```diff
@@ -62,12 +62,8 @@ bool FailureDetector::update(const vehicle_status_s &vehicle_status)
 		_status &= ~(FAILURE_ROLL | FAILURE_PITCH | FAILURE_ALT | FAILURE_EXT);
 	}
 
-	if (_esc_status_sub.updated()) {
```

Comment:
So the difference here is that we run it on every iteration, even if there is no status update. Will that cause an issue, maybe if the first update hasn't arrived yet?

Reply:
FYI This question is still valid since even though this line got changed it was only the naming.
Answer: I did not remove the condition on the esc_status being updated but I moved it into the `updateEscsStatus()` function such that consistently `updateXXX()` gets called for all the checks and the check itself decides what to depend on e.g. topic update. See https://github.com/PX4/Firmware/pull/15457/files#diff-84f12e22ed352e6c0b6b55e22b6f0628R160